azbus/receiver.go

```diff
@@ -240,20 +240,24 @@ func (r *Receiver) receiveMessages(ctx context.Context) error {
 				select {
 				case <-rctx.Done():
 					rr.log.Debugf("Stop worker %d", ii)
+					wg.Done()
 					return
 				case msg := <-msgs:
-					var renewCtx context.Context
-					var renewCancel context.CancelFunc
-					var maxDuration time.Duration
-					if rr.Cfg.RenewMessageLock {
-						renewCtx, renewCancel = context.WithCancel(rctx)
-						go rr.renewMessageLock(renewCtx, ii+1, msg)
-					} else {
-						// we need a timeout if RenewMessageLock is disabled
-						renewCtx, renewCancel, maxDuration = rr.setTimeout(rctx, rr.log, msg)
-					}
-					rr.processMessage(renewCtx, ii+1, maxDuration, msg, rr.handlers[ii])
-					renewCancel()
+					func(rrctx context.Context) {
```

Comment:
This looks both better and safer, but did something actually go wrong with the previous implementation ?

Reply:
Ah - the linter complained about too deep nested contexts

golangci-lint has version 1.60.3 built with go1.23.0 from c2e095c0 on 2024-08-22T21:51:56Z
azbus/receiver.go:249:7: nested context in loop (fatcontext)

See https://github.com/Crocmagnon/fatcontext
						renewCtx, renewCancel = context.WithCancel(rctx)
						^
task: Failed to run task "precommit": exit status 1

Reply:
if something was wrong with previous implementation we certainly did not notice